rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/toolWindow/log/UnityLogTokenizer.kt

```diff
@@ -0,0 +1,205 @@
+//                    break
+//                }
+            }
+        }
+
+        tokens.add(Token(fullString.substring(lastTokenIndex), UnityLogTokenType.String))
+
+        generateStyles(tokens)
+
+        return tokens
+    }
+
+    private fun generateStyles(tokens: MutableList<Token>) {
+        for ((i, token) in tokens.withIndex()) {
+            if (token.type == UnityLogTokenType.Bold) {
+                for (x in i until tokens.count()) {
+                    if (tokens[x].type == UnityLogTokenType.BoldEnd && !tokens[x].used) {
+                        token.used = true
+                        tokens[x].used = true
+
+                        for (y in i until x) {
+                            tokens[y].bold = true
+                        }
+                        break
+                    }
+                }
+            }
+            else if (token.type == UnityLogTokenType.Italic) {
+                for (x in i until tokens.count()) {
+                    if (tokens[x].type == UnityLogTokenType.ItalicEnd && !tokens[x].used) {
+                        token.used = true
+                        tokens[x].used = true
+
+                        for (y in i until x) {
+                            tokens[y].italic = true
+                        }
+                        break
+                    }
+                }
+            }
+            else if (token.type == UnityLogTokenType.Color) {
+                for (x in tokens.count() - 1 downTo i) {
+                    if (tokens[x].type == UnityLogTokenType.ColorEnd && !tokens[x].used) {
+                        token.used = true
+                        tokens[x].used = true
+
+                        var color = this.parseColor(getTokenValue(token.token))
+                        for (y in i until x) {
+                            tokens[y].color = color
+                        }
+                        break
+                    }
+                }
+            }
+            else if(validTokens.containsValue(token.type))
+            {
+                token.used = true
+            }
+        }
+    }
+
+    private fun addTokens(i: Int, lastTokenIndex: Int, tokens: MutableList<Token>, fullString: String, tokenString: String, type: UnityLogTokenType) {
+        if (i > lastTokenIndex)
+            tokens.add(Token(fullString.substring(lastTokenIndex, i), UnityLogTokenType.String))
+        tokens.add(Token(tokenString, type))
+    }
+
+    private fun getTokenValue(tokenString: String) : String
+    {
+        if(!tokenString.contains('='))
+            return ""
+
+        val cleanedToken = tokenString.replace(">", "")
+        return cleanedToken.substring(cleanedToken.indexOf('=') + 1).trim('"')
+    }
+
+    private fun checkToken(fullString: String, expectedToken: String, startIndex: Int): Int {
+        var expectedTokenIndex = 0
+
+        for (i in startIndex until fullString.length) {
+            val expectedChar = expectedToken[expectedTokenIndex]
```

Comment:
Does Unity treat token names as case insensitive? This only handles lowercase tokens